sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -510,39 +526,44 @@ def __init__(self, **kwargs: Any) -> None:
         self._update_lock = Lock()
         self._refresh_lock = Lock()
 
-    def refresh(self, **kwargs) -> None:
+    def refresh(self, **kwargs) -> None:  # pylint: disable=too-many-statements
         if not self._refresh_on and not self._feature_flag_refresh_enabled:
-            logging.debug("Refresh called but no refresh enabled.")
+            logger.debug("Refresh called but no refresh enabled.")
             return
         if not self._refresh_timer.needs_refresh():
-            logging.debug("Refresh called but refresh interval not elapsed.")
+            logger.debug("Refresh called but refresh interval not elapsed.")
             return
         if not self._refresh_lock.acquire(blocking=False):  # pylint: disable= consider-using-with
-            logging.debug("Refresh called but refresh already in progress.")
+            logger.debug("Refresh called but refresh already in progress.")
             return
         success = False
         need_refresh = False
         error_message = """
                         Failed to refresh configuration settings from Azure App Configuration.
                         """
         exception: Exception = RuntimeError(error_message)
+        is_failover_request = False
         try:
             self._replica_client_manager.refresh_clients()
-            active_clients = self._replica_client_manager.get_active_clients()
+            self._replica_client_manager.find_active_clients()
+            client_count = self._replica_client_manager.get_client_count() - 1
+
+            while client := self._replica_client_manager.get_next_active_client():
+                headers = _update_correlation_context_header(
+                    kwargs.pop("headers", {}),
+                    "Watch",
+                    client_count - 1,
```

Comment:
Dont need to subtract 1 here. Also, we can rename the variable to `replica_count` so that its clear why we're subtracting 1.

Reply:
In a previous comment you said we needed to subtract 1 as we only provide the discovered replica count.

Reply:
Just realized we subtract 1 twice, nvm.

Reply:
@avanigupta  Just to verify. Are we giving the total count of stores we have clients to, or the number of discovered replicas?

Reply:
This metric tracks the number of replicas a store is using. That's total number of clients at any time minus 1 (origin store).